client/question/components/LearnerViewDirective.js

```diff
@@ -737,13 +737,14 @@ tie.directive('learnerView', [function() {
           // or SessionResumeEvent, respectively.
           var hiddenAttributeName = (
             $scope.determineHiddenAttributeNameForBrowser());
-          if (hiddenAttributeName !== null) {
+          if (hiddenAttributeName !== null && tasks !== null &&
+              tasks.length > currentTaskIndex) {
```

Comment:
Why all these extra conditions?

Reply:
In case the question isn't fully initialized when a user switches tabs.

Just trying to be robust.

Reply:
OK. Is it possible that this might lead to an extra SessionResumeEvent (with no corresponding SessionPauseEvent)?

Reply:
I suppose that it's possible. What should we do about it?

Reply:
Discussed offline. It's probably fine; we can check later to see if there are any event discrepancies. Marking this as resolved.